test/AxelarGatewaySinglesig.js

```diff
@@ -1,2017 +1,1198 @@
-                  ['string', 'string', 'uint8', 'uint256', 'address'],
-                  [name, symbol, decimals, cap, ADDRESS_ZERO],
-                ),
-              ],
-            ],
-          ),
-        );
-
-        return getSignedExecuteInput(data, operatorWallet).then((input) =>
-          expect(contract.execute(input))
-            .to.not.emit(contract, 'TokenDeployed')
-            .and.to.not.emit(contract, 'Executed'),
-        );
-      });
-
-      it('should deploy a new token', () => {
-        const commandID = getRandomID();
-        const data = arrayify(
-          defaultAbiCoder.encode(
-            ['uint256', 'uint256', 'bytes32[]', 'string[]', 'bytes[]'],
-            [
-              CHAIN_ID,
-              ROLE_OWNER,
-              [commandID],
-              ['deployToken'],
-              [
-                defaultAbiCoder.encode(
-                  ['string', 'string', 'uint8', 'uint256', 'address'],
-                  [name, symbol, decimals, cap, ADDRESS_ZERO],
-                ),
-              ],
-            ],
-          ),
-        );
-
-        const tokenFactory = new ContractFactory(
-          BurnableMintableCappedERC20.abi,
-          BurnableMintableCappedERC20.bytecode,
-        );
-        const { data: tokenInitCode } = tokenFactory.getDeployTransaction(
-          name,
-          symbol,
-          decimals,
-          cap,
-        );
-        const expectedTokenAddress = getCreate2Address(
-          contract.address,
-          id(symbol),
-          keccak256(tokenInitCode),
-        );
-
-        return getSignedExecuteInput(data, ownerWallet)
-          .then((input) =>
-            expect(contract.execute(input))
-              .to.emit(contract, 'TokenDeployed')
-              .and.to.emit(contract, 'Executed')
-              .withArgs(commandID),
-          )
-          .then(() => contract.tokenAddresses(symbol))
-          .then((tokenAddress) => {
-            expect(tokenAddress).to.be.properAddress;
-            expect(tokenAddress).to.eq(expectedTokenAddress);
-
-            const tokenContract = new Contract(
-              tokenAddress,
-              BurnableMintableCappedERC20.abi,
-              ownerWallet,
-            );
+    describe('upgrade', () => {
+        it('should allow the admins to upgrade to the correct implementation', async () => {
+            const newGatewayImplementation = await gatewayFactory.deploy(tokenDeployer.address).then((d) => d.deployed());
+            const newGatewayImplementationCode = await newGatewayImplementation.provider.getCode(newGatewayImplementation.address);
+            const newGatewayImplementationCodeHash = keccak256(newGatewayImplementationCode);
+
+            const newAdminAddresses = getAddresses(admins.slice(0, 2));
+
+            const params = getSinglesigProxyDeployParams(newAdminAddresses, 2, wallets[5].address, wallets[6].address);
+
+            for (let i = 0; i < threshold - 1; ++i) {
```

Comment:
This should be done with `Promise.all` instead of looping.

Reply:
Fixed